iosHyperskillApp/iosHyperskillApp/Sources/Modules/Profile/ProfileViewModel.swift

```diff
@@ -165,20 +165,15 @@ final class ProfileViewModel: FeatureViewModel<
             let permissionStatus = await NotificationPermissionStatus.current
 
             if permissionStatus == .denied {
-                await MainActor.run {
-                    handleUserDeclinedDailyStudyReminders(shouldEmitObjectChange: true)
-                }
+                await handleUserDeclinedDailyStudyReminders()
             }
         }
     }
 
-    private func handleUserDeclinedDailyStudyReminders(shouldEmitObjectChange: Bool = false) {
-        notificationInteractor.setDailyStudyRemindersEnabled(enabled: false)
+    @MainActor
+    private func handleUserDeclinedDailyStudyReminders() async {
+        try? await notificationInteractor.setDailyStudyRemindersEnabled(enabled: false)
```

Comment:
Support for calling `suspend` functions from Swift code as `async` is [Experimental](https://kotlinlang.org/docs/components-stability.html). It may be dropped or changed at any time. Use it only for evaluation purposes.